dubs/dubs.py

```diff
@@ -0,0 +1,280 @@
+
+    def worker(self, entry, pdbid):
+        # Define and assign the reference pdbid
+        refpdbid = ""
+        # mode is 0 unassigned, 1 for alignment for protein, 2 for alignment for ligand
+        mode = 0
+
+        # Check for pdbID as a protein to be aligned (like in PINC)
+        for key, value in alignProtDict.items():
+            if pdbid in value:
+                refpdbid = key
+                mode = 1
+
+        # Check for protein-ligand pair for alignment 
+        for key, value in referenceDict.items():
+            if pdbid in value:
+                refpdbid = key
+                mode = 2
+
+        if mode == 0:
+
+            for ligand_code in pdbIDSMDict.get(pdbid, []):
+
+                ligand_ids = lemon.select_specific_residues(entry, lemon.ResidueName(ligand_code))
+                lemon.prune_identical_residues(entry, ligand_ids)
+
+                for ligand_id in ligand_ids:
+                    protein = lemon.Frame()
+                    ligand = lemon.Frame()
+
+                    lemon.separate_protein_and_ligand(entry, ligand_id, 25.0, protein, ligand)
+                    lemon.write_file(protein, pdbid + "_" + ligand_code + ".pdb")
+                    lemon.write_file(ligand, pdbid + "_" + ligand_code + ".sdf")
+
+            return pdbid + " no alignment"
+
+        elif mode == 1:
+            # If we need to align to a protein (like in PINC)
+            alignment = lemon.TMscore(entry, self.reference_structures[refpdbid])
+            positions = entry.positions()
+            lemon.align(positions, alignment.affine)
+            return "Align Protein: " + pdbid + " to " + refpdbid + " with score of " + alignment.score
+
+        elif mode == 2:
+            # If we are doing ligand alignment, that can be done here
+            # Get a list of the ligands associated with the protein we are trying to align
+            SM_ligandList = pdbIDSMDict.get(pdbid, []) 
+            Non_SM_ligandList = pdbIDNonSMDict.get(pdbid, [])
+
+            alignment = lemon.TMscore(entry, self.reference_structures[refpdbid])
+            positions = entry.positions()
+            lemon.align(positions, alignment.affine)
+
+            if len(SM_ligandList) > 0:
+
+                for ligand_code in SM_ligandList:
+
+                    ligand_ids = lemon.select_specific_residues(entry, lemon.ResidueName(ligand_code))
+                    lemon.prune_identical_residues(entry, ligand_ids)
+
+                    for ligand_id in ligand_ids:
+                        protein = lemon.Frame()
+                        ligand = lemon.Frame()
+
+                        lemon.separate_protein_and_ligand(entry, ligand_id, 25.0, protein, ligand)
+                        lemon.write_file(protein, pdbid + "_" + ligand_code + ".pdb")
+                        lemon.write_file(ligand, pdbid + "_" + ligand_code + ".sdf")
+
+            if len(Non_SM_ligandList > 0):
+                # TODO stuff here for non-small ligands
```

Comment:
Is this TBD before merging?

Reply:
Yes, we don't need it to support the benchmarks we claim to support